### PR TITLE
fix(backend-proxy-middleware): add url to type DestinationBackendConfig

### PR DESCRIPTION
**Pull request by zdravko-georgiev**
Follow up of https://github.com/SAP/open-ux-tools/pull/576

### CODE REVIEW

**Review by tobiasqueck — APPROVED** [review]
Thank you.

File: packages/backend-proxy-middleware/src/base/proxy.ts [hunk 1]
```diff
@@ -332,7 +332,7 @@ export async function generateProxyMiddlewareOptions(
     }
 
     backend.proxy = getCorporateProxyServer(backend.proxy);
-    if (backend.proxy && !isHostExcludedFromProxy(proxyOptions.target)) {
+    if (backend.proxy && !isHostExcludedFromProxy(proxyOptions.target!)) {
```
Comment (tobiasqueck) [inline packages/backend-proxy-middleware/src/base/proxy.ts:335]: Looking at this again, after responding to @devinea , I think this is incorrect. It is not guaranteed that target is defined.

File: packages/backend-proxy-middleware/src/base/types.ts [hunk 2]
```diff
@@ -33,6 +33,10 @@ export interface BaseBackendConfig {
 }
 
 export interface DestinationBackendConfig extends BaseBackendConfig {
+    /**
+     * Mandatory URL pointing to the backend system
+     */
+    url?: string;
```
[the remark this answers is not in the document]
Comment (zdravko-georgiev) [reply packages/backend-proxy-middleware/src/base/types.ts:39]: Yes, copy and paste error, I will change the comment

File: packages/backend-proxy-middleware/src/base/proxy.ts (hunk 1 above)
Comment (zdravko-georgiev) [reply packages/backend-proxy-middleware/src/base/proxy.ts:335]: Yup, I missed that `getDestinationUrlForAppStudio` can return `undefined`.

Comment (tobiasqueck) [reply packages/backend-proxy-middleware/src/base/proxy.ts:335]: But if that happens, the proxy cannot work, correct? So if proxyOptions.target is undefined at that point in time we should throw an error.

Comment (zdravko-georgiev) [reply packages/backend-proxy-middleware/src/base/proxy.ts:335]: Yes, without a `target` the proxy can't work.

File: packages/backend-proxy-middleware/src/base/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zdravko-georgiev) [reply packages/backend-proxy-middleware/src/base/types.ts:39]: Done with https://github.com/SAP/open-ux-tools/pull/581/commits/4ad8951f302d05ab47b993d22f5b2d33fbffcb73

File: packages/backend-proxy-middleware/src/base/proxy.ts (hunk 1 above)
Comment (zdravko-georgiev) [reply packages/backend-proxy-middleware/src/base/proxy.ts]: Done with https://github.com/SAP/open-ux-tools/pull/581/commits/4ad8951f302d05ab47b993d22f5b2d33fbffcb73

**Review by tobiasqueck — DISMISSED** [review]

**Review by tobiasqueck — APPROVED** [review]

**Review by Klaus-Keller — APPROVED** [review]
Thanks @zdravko-georgiev, 
code looks good and unit test exist.

**Pull request merged**